### PR TITLE
Add Records container class

### LINKED ISSUE #48: Add a `RecordsList` class?
Currently, when a slot has multiple records (e.g. `artifact$cell_types`) they are stored in a list. This works fine except that if a user access this slot the list is printed which can have many items if the records are e.g. genes. Adding a collection class would allow us to control the printing. This could be a full `R6` class or an `S3` extension of a list.

### PR DESCRIPTION
**Pull request by lazappi**
Adds a class for managing multiple records. I called it `RelatedRecords` because that's what it's used for and that is closer to the Python object.

Fixes #48

### CODE REVIEW

**Comment by lazappi** [comment]
Work in progress. Requires features from #54 so waiting for that to be merged first.

File: R/RelatedRecords.R [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      )
+    }
+  ),
+  active = list(
+    #' @field field ([Field])\cr
+    #' The field the records are related to.
+    field = function() {
+      private$.field
+    }
+  ),
```
Comment (rcannood) [inline R/RelatedRecords.R:70]: Why expose this interface? Does the Python interface also expose this information?

File: R/Record.R [hunk 2]
```diff
@@ -148,43 +148,57 @@ Record <- R6::R6Class( # nolint object_name_linter
     .api = NULL,
     .data = NULL,
     get_value = function(key) {
```
Comment (rcannood) [inline R/Record.R:150]: I changed the ordering of this function because I was starting to find it difficult to read

**Review by rcannood — COMMENTED** [review]
Great PR! I have a few suggestions:

* Could you update `architecture.qmd` to add the `$df()` methods and the relatedrecords class?

File: R/RelatedRecords.R [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    get_records = function(as_df = FALSE) {
+      field <- private$.field
+
+      # Fetch the field to get the related data
+      related_data <- private$.api$get_record(
+        module_name = field$module_name,
+        registry_name = field$registry_name,
+        id_or_uid = private$.related_to,
+        select = field$field_name,
+        limit_to_many = 100000L # Make this high to get all related records
```
[the remark this answers is not in the document]
Comment (lazappi) [reply R/RelatedRecords.R:86]: I didn't because Python doesn't have a limit argument but it would be easy enough to add

File: R/Record.R (hunk 2 above)
Comment (lazappi) [reply R/Record.R:150]: 👍🏻 I think I just didn't want to mess with the existing structure but this is better.

File: R/RelatedRecords.R [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      private$.related_to <- related_to
+    },
+    #' @description
+    #' Get a data frame summarising records in the registry
+    #'
+    #' @param limit Maximum number of records to return
+    #' @param verbose Boolean, whether to print progress messages
+    #'
+    #' @return A data.frame containing the available records
+    df = function(limit = 100, verbose = FALSE) {
```
Comment (rcannood) [inline R/RelatedRecords.R:33]: Is there code code that is similar in record$df vs. Relatedrecords$df?

File: R/RelatedRecords.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (lazappi) [reply R/RelatedRecords.R:86]: Not sure if we want to add this or not?

[the remark this answers is not in the document]
Comment (rcannood) [reply R/RelatedRecords.R:79]: We can leave it as is for now ^^

**Review by rcannood — APPROVED** [review]
LGTM! Thanks Luke!

**Pull request merged**